adapters/repos/db/vector/hnsw/search.go

```diff
@@ -32,23 +32,23 @@ func reasonableEfFromK(k int) int {
 	return ef
 }
 
-func (h *hnsw) SearchByID(id int, k int) ([]int, error) {
+func (h *hnsw) SearchByID(id int64, k int) ([]int64, error) {
```

Comment:
whole file: see above, all node ids should be represented by `uint64`, removing the need for the conversions.